src/feature.tf

```diff
@@ -0,0 +1,20 @@
+locals {
+  token_response = jsondecode(data.http.token.response_body)
+  token          = lookup(local.token_response, "access_token", "")
+}
+
+data "http" "token" {
+  url    = "https://login.microsoftonline.com/${var.azure_service_principal.data.tenant_id}/oauth2/token"
+  method = "POST"
+
+  request_body = "grant_type=Client_Credentials&client_id=${var.azure_service_principal.data.client_id}&client_secret=${var.azure_service_principal.data.client_secret}&resource=https://management.azure.com/"
+}
+
+data "http" "feature" {
```

Comment:
I think the cluster resource needs a `depends_on` for this data resource. That way it will wait for this to complete before trying to create the cluster.